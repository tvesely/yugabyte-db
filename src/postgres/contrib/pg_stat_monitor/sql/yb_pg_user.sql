CREATE USER su WITH SUPERUSER;

SET ROLE su;
CREATE EXTENSION pg_stat_monitor;

CREATE USER u1;
CREATE USER u2;

SET ROLE su;

SELECT pg_stat_monitor_reset();
SET ROLE u1;
CREATE TABLE t1 (a int);
SELECT * FROM t1;

-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: pg_stat_monitor parsing logic has run to run variability for parsing users.
-- SET ROLE u2;
-- CREATE TABLE t2 (a int);
-- SELECT * FROM t2;

SET ROLE su;
SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT pg_stat_monitor_reset();

DROP TABLE t1;
DROP TABLE t2;

DROP USER u1;
DROP USER u2;

--
-- create / alter user
--
SELECT pg_stat_monitor_reset();
CREATE USER foo PASSWORD 'foo';
ALTER USER foo PASSWORD 'foo2';
CREATE ROLE bar PASSWORD 'bar';
ALTER ROLE bar PASSWORD 'bar2';

SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";

DROP USER foo;
DROP ROLE bar;

DROP EXTENSION pg_stat_monitor;
