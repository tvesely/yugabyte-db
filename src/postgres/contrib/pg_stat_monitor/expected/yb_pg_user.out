CREATE USER su WITH SUPERUSER;
SET ROLE su;
CREATE EXTENSION pg_stat_monitor;
CREATE USER u1;
CREATE USER u2;
SET ROLE su;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

SET ROLE u1;
CREATE TABLE t1 (a int);
SELECT * FROM t1;
 a
---
(0 rows)

-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: pg_stat_monitor parsing logic has run to run variability for parsing users.
-- SET ROLE u2;
-- CREATE TABLE t2 (a int);
-- SELECT * FROM t2;
SET ROLE su;
SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
 userid |                                 query
--------+-----------------------------------------------------------------------
 u1     | CREATE TABLE t1 (a int);
 u1     | SELECT * FROM t1;
 su     | SELECT pg_stat_monitor_reset();
 su     | SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
 su     | SET ROLE su;
 u1     | SET ROLE u1;
(6 rows)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

DROP TABLE t1;
DROP TABLE t2;
DROP USER u1;
DROP USER u2;
--
-- create / alter user
--
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

CREATE USER foo PASSWORD 'foo';
ALTER USER foo PASSWORD 'foo2';
CREATE ROLE bar PASSWORD 'bar';
ALTER ROLE bar PASSWORD 'bar2';
SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
 userid |                                 query
--------+-----------------------------------------------------------------------
 su     | ALTER ROLE bar PASSWORD <REDACTED>
 su     | ALTER USER foo PASSWORD <REDACTED>
 su     | CREATE ROLE bar PASSWORD <REDACTED>
 su     | CREATE USER foo PASSWORD <REDACTED>
 su     | SELECT pg_stat_monitor_reset();
 su     | SELECT userid, query FROM pg_stat_monitor ORDER BY query COLLATE "C";
(6 rows)

DROP USER foo;
DROP ROLE bar;
DROP EXTENSION pg_stat_monitor;
